TLM/TLM/Util/Extensions/NetSegmentExtensions.cs

```diff
@@ -72,6 +72,41 @@ public static class NetSegmentExtensions {
             return new GetSegmentLaneIdsEnumerable(initialLaneId, netInfo.m_lanes.Length, laneBuffer);
         }
 
+        /// <summary>
```

Comment:
[Note]
Based this code `laneType` or `VehicleType` should never be 0. Maybe we can Assert this condition even if it is only in DEBUG build?
Is there a `[NotNull]` or `[NotZero]` annotation we can use here?

in some places in CS they check for`laneType == 0 || (laneInfo.m_laneType & laneType) != 0)` but I I think in speed limit manager that does not make sense.

Reply:
never mind. I deleted message.

Reply:
[low priority]
is this critical code path. do the assertion need to be debug only?

Reply:
> [low priority]
> is this critical code path. do the assertion need to be debug only?

heh, it was your suggesstion in https://github.com/CitiesSkylinesMods/TMPE/pull/1467#discussion_r834246755 :
> Maybe we can Assert this condition even if it is only in DEBUG build?